cmake/therock_subproject.cmake

```diff
@@ -615,145 +615,171 @@ function(therock_cmake_subproject_activate target_name)
     set(_all_option "ALL")
   endif()
 
-  # configure target
-  if(THEROCK_VERBOSE)
-    message(STATUS "  CONFIGURE_DEPENDS: ${_transitive_configure_depend_files} ")
-  endif()
+  # Detect whether the stage dir has been pre-built.
+  set(_prebuilt_file "${_stage_dir}.prebuilt")
   set(_configure_stamp_file "${_stamp_dir}/configure.stamp")
-  set(_configure_comment_suffix " (in background)")
-  set(_terminal_option)
-  set(_build_terminal_option "USES_TERMINAL")
-  if("$ENV{THEROCK_INTERACTIVE}")
-    set(_terminal_option "USES_TERMINAL")
-    set(_configure_comment_suffix)
-  elseif(_build_pool)
+  set(_build_stamp_file "${_stamp_dir}/build.stamp")
+  set(_stage_stamp_file "${_stamp_dir}/stage.stamp")
+
+  if(EXISTS "${_prebuilt_file}")
+    # If pre-built, just touch the stamp files, conditioned on the prebuilt
+    # marker file (which may just be a stamp file or may contain a unique hash
+    # for this part of the build).
+    message(STATUS "  DISABLING BUILD: Marked as pre-built")
```

Comment:
> * Updates the subproject CMake support to detect when this has been done and null out sub-project configure/build/stage steps for so populated entities.

I wonder if we can use this on Windows to support prebuilt CLR using `amdhip64_6.dll` from the existing public releases, while the open sourcing of PAL [static libraries] is still ongoing. Slotting into this generic "prebuilt" system that works for _all_ subprojects at least seems nicer than carving out a new prebuilt CLR subproject that could be selected instead of the source build.

Reply:
If you do that you'll end up with some kernels aot compiled with the in tree toolchain and others etc compiled with the blob version. Not saying it isn't tractable, but it will make the job of actually assembling a known performance setup on new configurations quite hard to navigate.